design/proof-network.md

```diff
@@ -0,0 +1,285 @@
+Storage proof network
+=====================
+
+Authors: Codex Team
+
+In this document we explore a design for an off-chain network for validating
+[storage proofs][1]. Instead of checking each storage proof in a smart contract
+on-chain, we let the proof network check these proofs. Only when a proof is
+missing we go on-chain to enact slashing. The main goal of this exercise is to
+reduce the costs of submitting and validating proofs, which has shown to be a
+limiting factor for the profitability of storage providers and the scaling of
+the storage network, even when deploying on a [rollup][2] or [sidechain][3].
+
+[1]: proof-erasure-coding.md
+[2]: ../evaluations/rollups.md
+[3]: ../evaluations/sidechains.md
+
+Overview
+--------
+
+The main idea is that validators in the network sign off on messages containing
+correct storage proofs, and on messages that correctly indicate that a proof is
+missing. These validators deposit a stake on-chain. We consider a message to be
+validated by the network when a subset of the validators representing > 2/3 of
+the total network stake have signed off on it. The assumption here is that less
+than 1/3 of the validators are byzantine, meaning that the rest is online and
+following protocol.
+
+Roles in the network are:
+
+- provers: they are the storage providers that submit storage proofs
```

Comment:
If I may suggest it, rename "provers" to "storage providers" through the text. For someone who is learning about the mechanism, it is less one agent to remember.